src/views/ToDoList/ToDoList.test.tsx

```diff
@@ -0,0 +1,262 @@
+
+          const titleInput = toDoList.getByPlaceholderText('title');
+
+          fireEvent.change(titleInput, { target: { value: newTaskTitle } });
+
+          const button = toDoList.getByRole('button');
+
+          fireEvent.click(button);
+
+          const newTaskParagraph = await toDoList.findByText(newTaskTitle, {
+            selector: 'p',
+          });
+
+          const errorParagraph = toDoList.queryByText(
+            'Failed to catch tasks list',
+            { selector: 'p' },
+          );
+
+          expect(newTaskParagraph).toBeDefined();
+          expect(errorParagraph).toBeNull();
+        });
+      });
+      describe('if the Api call is unsuccessful', () => {
+        beforeEach(() => {
+          (toDosApi.createNewTask as Mock).mockRejectedValue(null);
+        });
+        it('should throw an error and not create task', async () => {
+          const toDoList = render(<ToDoList />);
+
+          const titleInput = toDoList.getByPlaceholderText('title');
+
+          fireEvent.change(titleInput, { target: { value: newTaskTitle } });
+
+          const button = toDoList.getByRole('button');
+
+          fireEvent.click(button);
+
+          const newTaskParagraph = toDoList.queryByText(newTaskTitle, {
+            selector: 'p',
+          });
+
+          const errorParagraph = await toDoList.findByText(
+            'Something went wrong when creating the task',
+            { selector: 'p' },
+          );
+
+          expect(newTaskParagraph).toBeNull();
+          expect(errorParagraph).toBeDefined();
+        });
+      });
+    });
+  });
+  describe('when user clicks delete button on specific task', () => {
+    beforeEach(() => {
+      (toDosApi.fetchToDos as Mock).mockResolvedValue(returnedTasksList);
+    });
+    describe('and the Api call is successful', () => {
+      beforeEach(() => {
+        (toDosApi.deleteTask as Mock).mockResolvedValue('200');
+      });
+      it('should delete specific task and leave other tasks intact', async () => {
+        const toDoList = render(<ToDoList />);
+
+        const secondTasksButton = await toDoList.findByTestId(
+          'task-2-delete-button',
+        );
+
+        fireEvent.click(secondTasksButton);
+
+        const firstTasksParagraph = await toDoList.findByText(task1.title, {
+          selector: 'p',
+        });
+        const thirdTasksParagraph = await toDoList.findByText(task3.title, {
+          selector: 'p',
+        });
+
+        const secondTasksParagraph = toDoList.queryByText(task2.title, {
+          selector: 'p',
+        });
+
+        expect(firstTasksParagraph).toBeDefined();
+        expect(thirdTasksParagraph).toBeDefined();
+        expect(secondTasksParagraph).toBeNull();
+      });
+    });
+    describe('and the Api call is unsuccessful', () => {
+      beforeEach(() => {
+        (toDosApi.deleteTask as Mock).mockRejectedValue('400');
+      });
+      it('should not delete the task and display error', async () => {
```

Comment:
- not deleting the task
- displaying an error
are two separate test cases and should have two separate its